en/services/mission.md

```diff
@@ -235,16 +239,89 @@ In more detail, the sequence of operations is:
 1. If no `MISSION_ACK` is received the operation will eventually timeout and may be retried (see [above](#timeout)).
 
 
-### Upload Partial Mission {#upload_partial}
+### Partial Mission Upload {#upload_partial}
 
-TBD
+The diagram below shows the communication sequence to upload a partial mission to a drone (assuming all operations succeed).
 
+> **Note** The message sequence is the same as a [full mission upload](#uploading_mission) except that it is triggered with a [MISSION_WRITE_PARTIAL_LIST](../messages/common.md#MISSION_WRITE_PARTIAL_LIST) instead of a [MISSION_COUNT](../messages/common.md#MISSION_COUNT).
+  An update is expected to be robust (succeed completely or fail and revert to original version).
 
-### Download Partial Mission {#download_partial}
+<span></span>
+> **Note** Unlike for full mission update, partial update is not required to be robust; individual commands can fail.
 
-TBD
+{% mermaid %}
+sequenceDiagram;
+    participant GCS
+    participant Drone
+    GCS->>Drone: MISSION_WRITE_PARTIAL_LIST
+    GCS->>GCS: Start timeout
+    Drone->>GCS: MISSION_REQUEST_INT (n)
+    GCS->>GCS: Start timeout
+    GCS-->>Drone: MISSION_ITEM_INT (n)
+    Drone->>GCS: MISSION_REQUEST_INT (n+1)
+    GCS->>GCS: Start timeout
+    GCS-->>Drone: MISSION_ITEM_INT (n+1)
+    Drone->>GCS: MISSION_ACK
+{% endmermaid %}
+
+In more detail, the sequence of operations is:
+1. GCS (client) sends [MISSION_WRITE_PARTIAL_LIST](../messages/common.md#MISSION_WRITE_PARTIAL_LIST) specifying the start and end index of items to overwrite.
+   - A [timeout](#timeout) must be started for the GCS to wait on the response from Drone (`MISSION_REQUEST_INT`).
+1. Drone (server) receives the message, and prepares to upload mission items.
```

Comment:
Not documented: What happens if the write is invalid? IE the system has 3 waypoints but the request is to add items 100-150?